hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -401,4 +408,35 @@ private static void inferAvroSchema(Configuration conf, LogicalType rowType) {
       conf.setString(FlinkOptions.SOURCE_AVRO_SCHEMA, inferredSchema);
     }
   }
+
+  /**
+   *
+   * @param conf The configuration
+   */
+  private static void validateSourceSchema(Configuration conf) {
+    final HoodieTableMetaClient metaClient = StreamerUtil.metaClientForReader(conf, HadoopConfigurations.getHadoopConf(conf));
+    final TableSchemaResolver schemaResolver = new TableSchemaResolver(metaClient);
+    final Schema requiredSchema = StreamerUtil.getSourceSchema(conf);
+    final Schema srcSchema;
+
+    try {
+      srcSchema = schemaResolver.getTableAvroSchema();
+    } catch (Exception e) {
+      LOG.warn("Skipping validation for requiredSchema as table avro schema could not be fetched", e);
+      return;
+    }
+
```

Comment:
I don't know why the schema mismatch for the queries, somehow you need a catalog to manage the table and schema altogether, query a table with same path but wrong schema is meaningless, and is the wrong schema hand-written by user each time? Why not just fetch the schema through the catalog?

Reply:
Yes, there will not be any schema mismatches if table definitions are managed via catalogs. However, we have use cases where tables are not managed via a catalog.

Reply:
@danny0405 Yeap, agree.

If you feel that this PR is not required, I'll close this then. I'll apply it into our internal version instead.